### PR TITLE
Adapt shortened tracks resolution method to work with AlCaReco event content

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

This is a follow-up to PR https://github.com/cms-sw/cmssw/pull/43613.
That PR introduced  a novel data-driven method to artificially shorten muon tracks to assess the tracking pT resolution of short tracks, relying on shortening the muon tracks by creating track candidates with new hit collections that have the outermost hits removed from the original collection.
This method is potentially very useful to assess the performance of Tracker Alignment in terms of track transverse momentum (e.g. several weak modes can affect the PT resolution with a dependence along the tracker radius or the beamline).
This PR adapts `SingleLongTrackProducer` to work also in the case of ALCARECO event data format. Additionally it implements the same workflow in a standalone mode (not dependent on DQM) adding in `Alignment/OfflineValidation` the module `ShortenedTrackValidation` which can be used for validation purposes after standalone track refits with target conditions. 

#### PR validation:

`scram b runtests_testTrackingResolution` runs fine. Also run private validation using the configuration committed in `Alignment/OfflineValidation`. 

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

N/A

### CODE REVIEW

**Comment by mmusich** [comment]
@cmsbuild, please test

File: RecoTracker/FinalTrackSelectors/plugins/SingleLongTrackProducer.cc [hunk 1]
```diff
@@ -108,21 +108,22 @@ void SingleLongTrackProducer::produce(edm::Event &iEvent, const edm::EventSetup
...
-          dRmin = dr;
+    // Long track needs to be close to a good muon (only if requested)
+    if (matchInDr > 0.) {
+      for (const auto &m : *muons) {
+        if (m.isTrackerMuon()) {
+          tMuon++;
+          reco::Track matchedTrack = *(m.innerTrack());
+          TLorentzVector pMatched(matchedTrack.px(), matchedTrack.py(), matchedTrack.pz(), matchedTrack.pt());
+          // match to general track in deltaR
+          double dr = pTrack.DeltaR(pMatched);
```
Comment (perrotta) [inline RecoTracker/FinalTrackSelectors/plugins/SingleLongTrackProducer.cc:119]: Using DeltaR2 could lead to some (tiny, here) perf improvement

**Comment by perrotta** [comment]
+1
- Unit test succeed

File: RecoTracker/FinalTrackSelectors/plugins/SingleLongTrackProducer.cc (hunk 1 above)
Comment (mmusich) [reply RecoTracker/FinalTrackSelectors/plugins/SingleLongTrackProducer.cc]: Thanks for the suggestion, implemented at https://github.com/cms-sw/cmssw/pull/43729/commits/3da22f135cede8946c5fb127de8f66bbd4905ba0

**Comment by mmusich** [comment]
please test

**Comment by mmusich** [comment]
> I find strange that the AlCa signature was not reset after the commit of the modification, that was subsequent to that signature: is there any problem in bot?

Wasn't it agreed to not reset all signatures if the changes are in a package under another umbrella? Perhaps this is now in production.

**Comment by smuzaffar** [comment]
> > I find strange that the AlCa signature was not reset after the commit of the modification, that was subsequent to that signature: is there any problem in bot?
> 
> Wasn't it agreed to not reset all signatures if the changes are in a package under another umbrella? Perhaps this is now in production.

this is true @mmusich  but I see that the commits which were forced pushed do touch `alca` files so bot should have reset the signatures. I am checking

**Comment by perrotta** [comment]
No, that's correct, and I was just forgetting it. 
The package RecoTracker/FinalTrackSelectors is only under reconstruction signature, see https://cms-sw.github.io/categories.html
Sorry for the noise.

**Comment by mmusich** [comment]
> forced pushed do touch alca files so bot should have reset the signatures. 

In case it helps, indeed I force-pushed, but did not change the actual content, just rebasing to another IB.

**Comment by mmusich** [comment]
@perrotta note that now your signature is needed again

**Comment by perrotta** [comment]
+alca

**Comment by jfernan2** [comment]
+1

**Comment by tjavaid** [comment]
+1

- spurious differences seen from unstable WF `12434.7` (open issue #39803 )

**Comment by rappoccio** [comment]
+1

**Pull request merged**